### PR TITLE
Asset Import: Remove the unneeded mesh node in the emfx node structure.

### PR DESCRIPTION
**Pull request by amzn-rhhong**
1. Add back the logic to ignore the non-bone node in the assImpBone importer.
2. Skip adding a mesh node to emfx structure unless it has a child joint or child mesh node.
3. The root node became the only node that holds the combined mesh data.
4. Update some comments 

Before:
Screenshot 2022-04-16 235404

After:
Screenshot 2022-04-16 234035

Tested with: jack, rin, pyro, a couple mixamo character.

### CODE REVIEW

**Review by lsemp3d — APPROVED** [review]

File: Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp [hunk 1]
```diff
@@ -366,7 +366,45 @@ namespace EMotionFX
                 auto mesh = azrtti_cast<const SceneDataTypes::IMeshData*>(it->second);
                 if (mesh)
                 {
-                    outMeshIndices.push_back(nodeIndex);
+                    outMeshIndices.emplace_back(nodeIndex);
+
+                    // Don't need to add mesh node except it is a parent of other joint / mesh node.
```
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp:371]: nit: it is -> if it is
nit: other -> another

Update: _Don't need to add a mesh node except if it is a parent of another joint / mesh node._

File: Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp [hunk 2]
```diff
@@ -366,7 +366,45 @@ namespace EMotionFX
...
+                    // Example:
+                    // joint_1
+                    //   |____transform
+                    //   |____mesh_1 (keep)
+                    //         |____transform
+                    //         |____mesh_2 (remove)
+                    //         |      |____transform
+                    //         |_______joint_2
+                    //                |____transform
+                    // emfx doesn't need to contain the "end-point" mesh node because mesh buffers ultimately stored in a single atom mesh.
```
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp:381]: nit: mesh buffers 'are' ultimately...

File: Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp [hunk 3]
```diff
@@ -366,7 +366,45 @@ namespace EMotionFX
...
+                    // NOTE: Joint and mesh node often has a transform node as the children. To correctly detect whether a mesh node have a joint
```
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp:382]: nit: Joint and mesh node often have a transform node as their children. To correctly detect whether a mesh node has a joint....

**Review by hultonha — APPROVED** [review]
Change looks good, I just have a few minor suggestions with the comments.

The other question is do we/could we add a test to verify this change and ensure we get the expected hierarchy? It seems like this would be something very useful to verify

File: Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp (hunk 1 above)
Comment (amzn-rhhong) [reply Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp:371]: fixed

File: Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp (hunk 2 above)
Comment (amzn-rhhong) [reply Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp:381]: fixed

File: Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp (hunk 3 above)
Comment (amzn-rhhong) [reply Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Actor/ActorBuilder.cpp:382]: fixed

**Comment by amzn-rhhong** [comment]
@hultonha Good question on the tests. The actor builder tests for emfx actually were deprecated a year ago during the atom mesh unified format effort and never been bought back to life. I tried to re-enable it but got a bunch of errors. I think we may have to address this issue in a different PR.

**Comment by hultonha** [comment]
> @hultonha Good question on the tests. The actor builder tests for emfx actually were deprecated a year ago during the atom mesh unified format effort and never been bought back to life. I tried to re-enable it but got a bunch of errors. I think we may have to address this issue in a different PR.

Okay interesting, might be worth making a follow-up task/issue for this, as this will definitely add value longer-term as we make these fixes/changes and want to ensure no other functionality has regressed

**Pull request merged**